src/mappings/entities/mapCustomClaimType.ts

```diff
@@ -0,0 +1,105 @@
+import { Codec } from '@polkadot/types/types';
+import { EventIdEnum, ModuleIdEnum, Portfolio, CustomClaimType } from '../../types';
+import { bytesToString, getNumberValue, getTextValue } from '../util';
+import { HandlerArgs } from './common';
+
+export const getPortfolio = async ({
+  identityId,
+  number,
+}: Pick<Portfolio, 'identityId' | 'number'>): Promise<Portfolio> => {
+  const portfolioId = `${identityId}/${number}`;
+
+  const portfolio = await Portfolio.get(portfolioId);
+
+  if (!portfolio) {
+    throw new Error(`Portfolio number ${number} not found for DID ${identityId}`);
+  }
+
+  return portfolio;
+};
+
+export const createCustomClaimType = (
+  attributes: Pick<CustomClaimType, 'id' | 'name' | 'identityId'>,
+  blockId: string
+): Promise<void> => {
+  const { id, name, identityId } = attributes;
+  return CustomClaimType.create({
+    id,
+    name,
+    identityId,
+    createdBlockId: blockId,
+    updatedBlockId: blockId,
+  }).save();
+};
+
+/**
+ * Creates a CustomClaimType if not present.
+ *
+ * @note - WARNING: This is needed when an CustomClaim is created with a CustomClaimType that doesn't exist. It should not be used unless necessary.
+ */
+export const createCustomClaimTypeIfNotExists = async (
+  { id }: Pick<CustomClaimType, 'id'>,
+  blockId: string
+): Promise<void> => {
+  const customClaimType = await CustomClaimType.get(id);
+
+  if (!customClaimType) {
+    await createCustomClaimType(
+      {
+        id,
+        name: '',
+        identityId: '',
+      },
+      blockId
+    );
+  }
+};
```

Comment:
I'm not sure why we need this. If the data is not there for old block, we need to use the migration handler to process all the old events and add entries for those and update the current data as well.

Also, creating a CustomClaimType with name and identity as blank would lead to misleading info for the user querying it.